### PR TITLE
Support for multiple correlation_id per single context

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: src/clients/c++/examples/simple_sequence_client.cc [hunk 1]
```diff
@@ -276,47 +275,48 @@ main(int argc, char** argv)
   std::vector<int32_t> result1_list;
 
   std::vector<std::unique_ptr<nic::InferContext>> ctxs;
```
Comment (deadeyegoodwin) [inline src/clients/c++/examples/simple_sequence_client.cc]: You can move this declaration down as well?

File: src/clients/c++/library/request.h.in [hunk 2]
```diff
@@ -528,6 +528,16 @@ class InferContext {
     /// \param batch_size The batch size.
     virtual void SetBatchSize(size_t batch_size) = 0;
 
+    /// \return The correlation id for the inferences.
+    virtual CorrelationID CorrelationId() const = 0;
+
+    /// Set the correlation id to use for subsequent inferences.
+    /// \param correlation_id The value of correlation id to be set.
+    /// CorrelationID of the context will only be modified if
```
Comment (deadeyegoodwin) [inline src/clients/c++/library/request.h.in]: Description is confusing. "If non-zero this correlation ID overrides the context's correlation ID for all subsequent inference requests".

File: src/clients/c++/examples/simple_sequence_client.cc (hunk 1 above)
Comment (tanmayv25) [reply src/clients/c++/examples/simple_sequence_client.cc]: Done

File: src/clients/c++/library/request.h.in (hunk 2 above)
Comment (tanmayv25) [reply src/clients/c++/library/request.h.in]: Fixed

**Pull request merged**